### PR TITLE
Add periodic test for backstage-showcase

### PR DESCRIPTION
**Pull request by josephca**
Add nightly test for the 'backstage-showcase/main'.

### CODE REVIEW

**Comment by josephca** [comment]
@subhashkhileri can you review the change? thanks!

File: ci-operator/step-registry/janus-idp/backstage-showcase/nightly/janus-idp-backstage-showcase-nightly-commands.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export HOME WORKSPACE
+HOME=/tmp
+WORKSPACE=$(pwd)
+cd /tmp || exit
+
+export GITHUB_ORG_NAME GITHUB_REPOSITORY_NAME NAME_SPACE TAG_NAME
+
+GITHUB_ORG_NAME="janus-idp"
+GITHUB_REPOSITORY_NAME="backstage-showcase"
+NAME_SPACE=showcase-ci
```
Comment (subhashkhileri) [inline ci-operator/step-registry/janus-idp/backstage-showcase/nightly/janus-idp-backstage-showcase-nightly-commands.sh:11]: This namespace might conflict with the PR check.
Can we update it to `showcase-ci-nightly`? wdyt?

File: ci-operator/jobs/janus-idp/backstage-showcase/janus-idp-backstage-showcase-main-presubmits.yaml [hunk 2]
```diff
@@ -13,7 +13,6 @@ presubmits:
     labels:
       ci.openshift.io/generator: prowgen
       pj-rehearse.openshift.io/can-be-rehearsed: "true"
-    max_concurrency: 1
```
Comment (subhashkhileri) [inline ci-operator/jobs/janus-idp/backstage-showcase/janus-idp-backstage-showcase-main-presubmits.yaml:16]: `max_concurrency: 1` This is for PR check to run serially. lets keep it. you need to manually update this file to add it.

**Review by subhashkhileri — CHANGES_REQUESTED** [review]

File: ci-operator/jobs/janus-idp/backstage-showcase/janus-idp-backstage-showcase-main-presubmits.yaml (hunk 2 above)
Comment (josephca) [reply ci-operator/jobs/janus-idp/backstage-showcase/janus-idp-backstage-showcase-main-presubmits.yaml:16]: Thanks for the catch. I didn't intend to change it.

**Comment by josephca** [comment]
@subhashkhileri I made the change for the review comments. Please review again. Thanks

File: ci-operator/config/janus-idp/backstage-showcase/janus-idp-backstage-showcase-main__nightly.yaml
Comment (subhashkhileri) [inline ci-operator/config/janus-idp/backstage-showcase/janus-idp-backstage-showcase-main__nightly.yaml:1]: @josephca I think we don't really need a new config file.
we can add it in existing config yaml. that is e2e-tests.

in that we can simply add 
tests:
- as: e2e-tests-nightly
  cron: 0 1 * * *
  steps:
    test:
    - ref: janus-idp-backstage-showcase-nightly
    
    
    and keep the step-registry separate for nightly tests.

**Comment by subhashkhileri** [comment]
/pj-rehearse periodic-ci-janus-idp-backstage-showcase-main-e2e-tests-nightly

**Comment by josephca** [comment]
@davidfestal , @invincibleJai - can you review and approve it? now the nightly test is passed. Thanks!

**Comment by davidfestal** [comment]
> @davidfestal , @invincibleJai - can you review and approve it? now the nightly test is passed. Thanks!

Could you rebase the PR and solve conflicts first ?

File: ci-operator/config/janus-idp/backstage-showcase/janus-idp-backstage-showcase-main__nightly.yaml
Comment (josephca) [reply ci-operator/config/janus-idp/backstage-showcase/janus-idp-backstage-showcase-main__nightly.yaml:1]: Done. Thanks,

**Comment by josephca** [comment]
> > @davidfestal , @invincibleJai - can you review and approve it? now the nightly test is passed. Thanks!
> 
> Could you rebase the PR and solve conflicts first ?
@davidfestal Done. Thanks!

**Pull request merged**